Landry333/Big-Owl
# #237 Notification stored per user instead.

### Important Notes:
- The original issue had an incorrect Issue Id, so the branch name and commit ids are incorrect. I'll make sure to be more careful about these things next time.
- Once this is merged, all Notifications in the base notification collection will become obselete

-----

### Related Issues
Closes #237 

### Description
When we add and query for notifications in the database, they are now specifically tied to and stored based on the receiver of that notification and not stored all together in one location, making queries more efficient. This has been done by making notifications sub collections of the user collection.

### How to check PR
- Login as any user
- Send a Supervision Request to any user already in the system
- check the database and you should see that the notification is stored under a user document, specifically the user that was given the supervision request
- if you login as the user you sent the request to, you should see the notifications

You can also check that other notification types are still working, and that the app behavior hasn't changed.

### Screenshot:

Here's a quick screenshot just to show what it looks like when notifications are under a user:
![image](https://user-images.githubusercontent.com/31963426/110561844-6513eb80-8116-11eb-816c-9fad6ea339e4.png)
